elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/TableContext.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.queryengines.sql;
+
+import static com.yahoo.elide.core.utils.TypeHelper.appendAlias;
+import static com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata.SQLReferenceTable.PERIOD;
+import static com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata.SQLReferenceTable.applyQuotes;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.dialects.SQLDialect;
+import com.github.jknack.handlebars.HandlebarsException;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * TableContext for Handlebars Resolution.
+ */
+@Getter
+@ToString
+@Builder
+public class TableContext extends HashMap<String, Object> {
+
+    private final String alias;
+    private final SQLDialect dialect;
+    private final Map<String, Object> defaultTableArgs;
+    @Builder.Default
+    private final Map<String, TableContext> joins = new HashMap<>();
```

Comment:
Instead of storing TableContext objects here for joins, I wonder if it would be clearer if we simply store the SQLTable directly in the context (which now also has a map of SQLJoin objects).  We are building a context object anyway each time.  It's not intuitive why the context copies are necessary - and reusing existing metadata concepts will remove this distinction.

Similarly, we might no longer need ColumnContext (because we have a reference to the underlying column).   My point is that we already have metadata objects - it might be clearer if we use them directly here.

While this suggestion may work nicely for tables, it may not work for nested queries.  

It might be good revisit the metadata design a bit here.  Let's discuss.